src/Octoshift/BaseCommandArgs.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+
+namespace OctoshiftCLI
+{
+    public abstract class BaseCommandArgs
+    {
+        [LogName("VERBOSE")]
+        public bool Verbose { get; set; }
+
+        public abstract void Validate();
+
+        public void Log(OctoLogger log)
+        {
+            if (log is null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            log.Verbose = Verbose;
+            var sb = new StringBuilder();
+
+            foreach (var property in GetType().GetProperties())
+            {
+                var logName = property.GetCustomAttributes(typeof(LogNameAttribute), true).FirstOrDefault() is LogNameAttribute logNameAttribute
+                        ? logNameAttribute.LogName
+                        : property.Name;
+
+                if (property.PropertyType == typeof(bool))
+                {
+                    if ((bool)property.GetValue(this))
+                    {
+                        sb.AppendLine($"{logName}: true");
+                    }
+                }
+                else
+                {
+                    sb.AppendLine($"{logName}: {property.GetValue(this)}");
+                }
+            }
+
+            log.LogInformation(sb.ToString());
+        }
+
+        public void RegisterSecrets(OctoLogger log)
+        {
+            foreach (var property in GetType().GetProperties())
+            {
+                if (property.GetCustomAttributes(typeof(SecretAttribute), true).Any())
+                {
+                    log?.RegisterSecret((string)property.GetValue(this));
```

Comment:
I guess like the `Log` method it'd be better to add a null guard for log and throw if it's null.

Reply:
done